### ISSUE TITLE
gcc-11.2 toolchain build for armv8-rpi3-linux-gnueabihf fails when configured to use glibc 2.31 

### ISSUE THREAD

**Issue by mbulut** [crosstool-ng/crosstool-ng#1655]
Hi,

my gcc-11.2 toolchain build for armv8-rpi3-linux-gnueabihf fails when I modify the glibc version to 2.31

```
[INFO ]    Building for multilib 1/1: ''
[EXTRA]      Configuring C library
[EXTRA]      Installing C library headers
[EXTRA]      Installing C library start files
[ERROR]      ../sysdeps/nptl/pthread.h:719:47: error: argument 1 of type 'struct __jmp_buf_tag *' declared as a pointer [-Werror=array-parameter=]
[ERROR]      ../sysdeps/nptl/pthread.h:719:47: error: argument 1 of type 'struct __jmp_buf_tag *' declared as a pointer [-Werror=array-parameter=]
[ERROR]      ../sysdeps/nptl/pthread.h:719:47: error: argument 1 of type 'struct __jmp_buf_tag *' declared as a pointer [-Werror=array-parameter=]
[ERROR]      make[3]: *** [../Makerules:271: /home/gollum/work/domino/venus/docker/.build/armv8-rpi3-linux-gnueabihf/build/build-libc-startfiles/multilib/tlsdesc.h] Error 1
[ERROR]      make[3]: *** Waiting for unfinished jobs....
[ERROR]      make[3]: *** [../Makerules:271: /home/gollum/work/domino/venus/docker/.build/armv8-rpi3-linux-gnueabihf/build/build-libc-startfiles/multilib/rtld-global-offsets.h] Error 1
[ERROR]      make[3]: *** [../Makerules:271: /home/gollum/work/domino/venus/docker/.build/armv8-rpi3-linux-gnueabihf/build/build-libc-startfiles/multilib/tcb-offsets.h] Error 1
[ERROR]      make[2]: *** [Makefile:470: csu/subdir_lib] Error 2
[ERROR]      make[1]: *** [Makefile:9: csu/subdir_lib] Error 2
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library headers & start files'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@383]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1608]
[ERROR]  >>        called from: glibc_backend[scripts/build/libc/glibc.sh@74]
[ERROR]  >>        called from: glibc_start_files[scripts/build/libc/glibc.sh@38]
[ERROR]  >>        called from: do_libc_start_files[scripts/build/libc.sh@28]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@697]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]   
[ERROR]  (elapsed: 5:36.33)
```

Without that modification, the build runs smoothly... Could not find any known issue with glibc 2.31, so have you guys ever encountered this?

host OS and version: Ubuntu 20.04.3 LTS

crosstool-NG version: current master (6737cfaa8ce629a6d35a5b56777c606b2712b630)

Toolchain config and build log attached:
[config.txt](https://github.com/crosstool-ng/crosstool-ng/files/7908573/config.txt)
[build.log](https://github.com/crosstool-ng/crosstool-ng/files/7908599/build.log)

**Comment by stilor** [comment]
https://sourceware.org/bugzilla/show_bug.cgi?id=26647

Newer GCC started to emit a warning (which due to -Werror in glibc's build flags are treated as errors); it eventually was fixed in newer glibc releases... Either need to pick up the patch and backport it to supported glibc releases, or add a dependency to disallow newer compiler versions for older glibc releases.

**Comment by mbulut** [comment]
@stilor what is the preferred way to hand crosstool-ng patches to apply to glibc after it has fetched the sources?

**Comment by krmao** [comment]
how to fix ? dest linux is glibc 2.31
